### PR TITLE
Adds a sesonal only machinecarbine gun, and a bucket for it

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Adds the ALF-51B machine carbine and the seasonal for it, type 51 slight description change for the seasonal since moths aren't real.

stats of the ALF51-B are 80 round magazines, same bullet as skirmisher, but it has triple falloff so it does nearly no damage past CQC, other than that it's just a T-18 with a big box magazine and worse gun stats and no aim mode, much less utiltiy in attachables.

image
it also has a firing animation, neat.

## Why It's Good For The Game
Adds a unique seasonal gun and the Type-71 deserved a seasonal for a bit.
I think the 51B will be a unique enough gun to warrant this, though it might need more tweaking

## Changelog
:cl:
add: machinecarbine, has the stats of a t-18, but with worse slowdown, incredibly high falloff, has a bigger damaging/sundering bullet, has more in the magazine. Also no aim mode.
expansion: Machinecarbine/Type-71 bucket for seasonals.
/:cl:

### CODE REVIEW

**Comment by Kuro020** [comment]
this is so ugly I'm crying

**Review by Kuro020 — APPROVED** [review]

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -1229,3 +1232,45 @@
 	wield_delay = 0.6 SECONDS
 	aim_slowdown = 0.5
 	damage_falloff_mult = 0.5
+
+//-------------------------------------------------------
+//AF-51B MACHINEcarbine
+
+/obj/item/weapon/gun/rifle/alf51b
+	name = "\improper ALF-51B machinecarbine"
+	desc = "The ALF-51B is an unoffical modification of a ALF-4, or better known as the T-18 carbine, modified to SMG length of barrel, rechambered for a stronger round, and belt based. Truly the peak of CQC. Useless past that. Aiming is impossible. Uses 10x25mm caseless ammunition."
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/rifles.dm:1241]: Dont use numbers in typepaths

File: code/modules/projectiles/magazines/rifles.dm [hunk 2]
```diff
@@ -342,3 +342,17 @@
 	max_rounds = 30
 	gun_type = /obj/item/weapon/gun/rifle/standard_skirmishrifle
 	icon_state_mini = "mag_rifle"
+
+//ALF-51B
+
+/obj/item/ammo_magazine/rifle/alf51b
```
Comment (TiviPlus) [inline code/modules/projectiles/magazines/rifles.dm:348]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest is fine

**Review by TiviPlus — APPROVED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**